### DISCUSSION TITLE
Penambahan Gallery Foto

### DISCUSSION THREAD

**Discussion by mharisudn** [buku-masjid/buku-masjid#58] [Ideas]
Penambahan galleri foto terutama jika ada progres pembangunan attribute masjid

**Comment by nafiesl** [comment]
mas @mharisudin setuju, ini fitur yang diperlukan juga. Berarti kemungkinanya seperti ini:

1. Pada halaman publik, saat melihat laporan buku kas, pengunjung melihat 1 tab tambahan yaitu Galeri Foto. Galeri foto ini berisi foto-foto yang diupload oleh admin, sekretaris, atau bendahara sesuai kegiatan setiap buku kas.

2. Pada halaman admin, kita tambahkan 1 menu lagi Galeri Foto. Pada menu tersebut ada fitur:
    a. Galeri foto berupa thumbnail diurutkan mulai tanggal upload paling baru (descending)
    b. Bisa upload foto baru
    c. Bisa menghapus foto
    d. Bisa edit judul dan deskripsi foto, dan visibilitas (internal/publik)
    e. Satu foto bisa dijadikan foto profil buku kas.

3. Fitur teknis yang diperlukan terkait galeri ini:
    a. Setiap upload foto, dilakukan resize agar dimensi lebar menjadi 1000px
    b. Setiap foto ada 2 file hasil resize: yang lebar 1000px dan thumbnail (200 x 200 px).
    c. Jika sistem menggunakan fitur queue laravel, maka proses resize dilakukan oleh job (queue)

Yang perlu didiskusikan:
1. Apakah kita develop fitur sendiri, atau pakai package yang sudah  ada.
2. Jika develop fitur sendiri, seperti apa struktur databasenya?
